route/v1/types.go

```diff
@@ -47,6 +47,7 @@ type Route struct {
 	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
 
 	// spec is the desired state of the route
+	// +kubebuilder:validation:XValidation:rule="!has(self.tls) || self.tls.termination != 'passthrough' || !has(self.httpHeaders)",message="header actions are not permitted when tls termination is passthrough."
```

Comment:
Is this a correct validation?  Shouldn't it be:
```suggestion
	// +kubebuilder:validation:XValidation:rule="self.tls.termination == 'passthrough' && has(self.httpHeaders)",message="header actions are not permitted when tls termination is passthrough."
```

Reply:
The validation is correct as it is, the validation rule will currently pass for a valid case, your suggested rule will fail on a valid case, which causes the message to be printed. If you check the integration tests that were added you can see we are testing this validation

Reply:
I've got it now, @Miciah explained it to me.  I thought it was `if rule is true then print message`, and it's not.